libultraship/libultraship/Window.cpp

```diff
@@ -149,13 +150,28 @@ extern "C" {
     }
 
     char* ResourceMgr_LoadTexByName(char* texPath) {
-        const auto res = static_cast<Ship::Texture*>(Ship::GlobalCtx2::GetInstance()->GetResourceManager()->LoadResource(texPath).get());
+        const auto res = LOAD_TEX(texPath);
         ModInternal::ExecuteHooks<ModInternal::LoadTexture>(texPath, &res->imageData);
         return (char*)res->imageData;
     }
 
+    uint16_t ResourceMgr_LoadTexWidthByName(char* texPath) {
+        const auto res = LOAD_TEX(texPath);
+        return res->width;
+    }
+
+    uint16_t ResourceMgr_LoadTexHeightByName(char* texPath) {
+        const auto res = LOAD_TEX(texPath);
+        return res->height;
+    }
+
+    uint32_t ResourceMgr_LoadTexSizeByName(char* texPath) {
+        const auto res = LOAD_TEX(texPath);
+        return res->imageDataSize;
+    }
+
```

Comment:
Would probably be good to have some kind of check to see if `res` is null, incase the provided path points to a non-existent resource.